FlingEngine/Graphics/src/Cubemap.cpp

```diff
@@ -66,126 +69,35 @@ namespace Fling
             m_UniformBuffers[i]->MapMemory();
         }
 
-
         UpdateUniformBuffer(t_CurrentImage, t_Camera->GetProjectionMatrix(), t_Camera->GetViewMatrix());
         SetupDescriptors();
-        PreparePipeline(t_SampleCount);
+        PreparePipeline(t_Sampler);
     }
 
-    void Cubemap::PreparePipeline(VkSampleCountFlagBits t_SampleCount)
+    void Cubemap::PreparePipeline(Multisampler* t_Sampler)
     {
-        VkPipelineInputAssemblyStateCreateInfo inputAssemblyState =
-            Initializers::PipelineInputAssemblyStateCreateInfo(
-                VK_PRIMITIVE_TOPOLOGY_TRIANGLE_LIST,
-                0,
-                VK_FALSE);
-
-        VkPipelineRasterizationStateCreateInfo rasterizationState =
-            Initializers::PipelineRasterizationStateCreateInfo(
-                VK_POLYGON_MODE_FILL,
-                VK_CULL_MODE_FRONT_BIT,
-                VK_FRONT_FACE_COUNTER_CLOCKWISE,
-                0);
-
-        VkPipelineColorBlendAttachmentState blendAttachmentState =
-            Initializers::PipelineColorBlendAttachmentState(
-                0xf,
-                VK_FALSE);
-
-        VkPipelineColorBlendStateCreateInfo colorBlendState =
-            Initializers::PipelineColorBlendStateCreateInfo(
-                1,
-                &blendAttachmentState);
-
-        VkPipelineDepthStencilStateCreateInfo depthStencilState =
-            Initializers::DepthStencilState(
-                VK_FALSE,
-                VK_FALSE,
-                VK_COMPARE_OP_LESS_OR_EQUAL);
-
-        VkPipelineViewportStateCreateInfo viewportState =
-            Initializers::PipelineViewportStateCreateInfo(
-                1, 1, 0);
-
-        VkPipelineMultisampleStateCreateInfo multisampleState =
-            Initializers::PipelineMultiSampleStateCreateInfo(
-                t_SampleCount,
-                0);
-
-
-        std::vector<VkDynamicState> dynamicStateEnables = {
-                 VK_DYNAMIC_STATE_VIEWPORT,
-                 VK_DYNAMIC_STATE_SCISSOR
-        };
-
-        VkPipelineDynamicStateCreateInfo dynamicState =
-            Initializers::PipelineDynamicStateCreateInfo(
-                dynamicStateEnables, 0);
+        Shader* vertShader = new Shader(m_VertexShader);
```

Comment:
Woah woah woah what's goin on here? We already have m_VertexShader as a `std::shadred_ptr`, just do `m_VertexShader.get()`